lib/foreman_inventory_upload/generators/slice.rb

```diff
@@ -105,8 +105,8 @@ def report_system_profile(host, host_ips_cache)
         @stream.array_field('network_interfaces') do
           @stream.raw(host.interfaces.map do |nic|
             {
-              'ipv4_addresses': [host_ips_cache[nic.ip]].compact,
-              'ipv6_addresses': [nic.ip6].compact,
+              'ipv4_addresses': [host_ips_cache[nic.ip]].reject(&:empty?),
+              'ipv6_addresses': [nic.ip6].reject(&:empty?),
```

Comment:
How would I go about getting a nic where the address would be an empty string like you have in tests?

Reply:
I suppose it comes from hosts that are created in foreman from imported facts. We have a customer that we have seen this behaviour. I can get more details, if you like.

Reply:
Please do